src/lazy/text/matched.rs

```diff
@@ -338,7 +338,14 @@ impl MatchedString {
         let body = matched_input.slice(3, matched_input.len() - 6);
         // There are no escaped characters, so we can just validate the string in-place.
         let mut sanitized = Vec::with_capacity(matched_input.len());
-        escape_text(body, &mut sanitized)?;
+        decode_text_containing_escapes(
+            body,
+            &mut sanitized,
+            // Normalize newlines
+            true,
+            // Support unicode escapes
+            true,
```

Comment:
🗺️ I considered enums for these two `bool`s to make them self-documenting, but as they're not part of the public API I decided to just comment the handful of places where this method is called.